### PR TITLE
Check DF flag in X86 signal handler

### PR DESCRIPTION
**Pull request by liqunl**
In the event of a seg fault in the JIT code, the DF flag should be
cleared, otherwise something must be wrong in the JIT code, in which
case we should fail the VM right away to generate a core dump with the
right context for defect investigation.

This commit adds such check which is to be enabled by an environment
variable in testing.

Signed-off-by: Liqun Liu <liqunl@ca.ibm.com>

### CODE REVIEW

**Comment by liqunl** [comment]
@gacholio @0xdaryl May I ask for your review?
I'm using `getenv` from `stdlib.h`, I wonder if we should use the one from portlib? If so, what header file I should include?

**Comment by gacholio** [comment]
Portlib would be preferred certainly - you should already have the right stuff included (if not, "j9port.h" is where you would find it).

File: runtime/compiler/runtime/SignalHandler.c [hunk 1]
```diff
@@ -31,6 +32,31 @@
...
+	U_32 infoType;
+	UDATA eflags;
+	const char* enableDFCheck = getenv("TR_trapIfDFSet");
+
+	if (enableDFCheck) {
+		infoType = j9sig_info(sigInfo, J9PORT_SIG_CONTROL, J9PORT_SIG_CONTROL_X86_EFLAGS, &infoName, &infoValue);
+		if (infoType == J9PORT_SIG_VALUE_ADDRESS) {
+			eflags = *(UDATA*)infoValue;
+			if ((eflags & 0x400) != 0) {
+				fprintf(stdout, "eflags %lx, eflags & 0x400 = %lx, DF flag is set\n", eflags, eflags & 0x400);
```
Comment (gacholio) [inline runtime/compiler/runtime/SignalHandler.c:50]: As a rule, the VM prints to stderr. If you use `j9tty_printf` you'll get this for free. UDATA-sized values are `%zx` in the port library printf.

**Comment by gacholio** [comment]
`getenv` is used in other places in the existing code, so there's no real need to use the port library here for debug code, as long as it compiles on all of the x86 platforms.

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins test sanity xlinux,win,osx jdk8

**Comment by gacholio** [comment]
For completeness, a compile on an unsupported platform.

**Comment by gacholio** [comment]
jenkins compile plinux jdk8

**Pull request merged**